invenio_files_rest/helpers.py

```diff
@@ -291,3 +292,13 @@ def create_file(key, path):
                 assert filename.startswith(source)
                 parts = [p for p in filename[len(source):].split(os.sep) if p]
                 yield create_file('/'.join(parts), os.path.join(root, name))
+
+
+def create_file_streaming_redirect_response(obj):
+    """Redirect response generating function."""
+    response = make_response()
+    redirect_url_base = '/secure_files/'
+    redirect_url_key = urlsplit(obj.file.uri).path
```

Comment:
just double check when the `obj` has a delete marker, if everything is ok, otherwise it might fail doing `obj.file`.